third_party/move/move-vm/runtime/src/move_vm.rs

```diff
@@ -88,108 +106,90 @@ impl MoveVM {
                     .clone(),
                 remote,
             ),
-            module_store: ModuleStorageAdapter::new(self.runtime.module_storage()),
+            module_store: ModuleStorageAdapter::new(self.runtime.module_storage_v1()),
             native_extensions,
         }
     }
 
-    /// Create a new session, as in `new_session`, but provide native context extensions and custome storage for resolved modules.
-    pub fn new_session_with_extensions_and_modules<'r>(
-        &self,
-        remote: &'r impl MoveResolver,
-        module_storage: Arc<dyn ModuleStorage>,
-        native_extensions: NativeContextExtensions<'r>,
-    ) -> Session<'r, '_> {
-        Session {
-            move_vm: self,
-            data_cache: TransactionDataCache::new(
-                self.runtime
-                    .loader()
-                    .vm_config()
-                    .deserializer_config
-                    .clone(),
-                remote,
-            ),
-            module_store: ModuleStorageAdapter::new(module_storage),
-            native_extensions,
-        }
-    }
-
-    /// Load a module into VM's code cache
+    /// DO NOT USE THIS API!
+    ///
+    /// Existing uses of this API is to fetch metadata from compiled modules on the client
+    /// side. With loader V2 design clients can fetch it directly from the module storage.
+    #[deprecated]
     pub fn load_module(
         &self,
         module_id: &ModuleId,
         remote: &impl MoveResolver,
     ) -> VMResult<Arc<CompiledModule>> {
-        self.runtime
-            .loader()
-            .load_module(
-                module_id,
-                &mut TransactionDataCache::new(
-                    self.runtime
-                        .loader()
-                        .vm_config()
-                        .deserializer_config
-                        .clone(),
-                    remote,
-                ),
-                &ModuleStorageAdapter::new(self.runtime.module_storage()),
+        match self.runtime.loader() {
+            Loader::V1(loader) => {
+                let module = loader.load_module(
+                    module_id,
+                    &mut TransactionDataCache::new(
+                        self.runtime
+                            .loader()
+                            .vm_config()
+                            .deserializer_config
+                            .clone(),
+                        remote,
+                    ),
+                    &ModuleStorageAdapter::new(self.runtime.module_storage_v1()),
+                )?;
+                Ok(module.as_ref().deref().clone())
+            },
+            Loader::V2(_) => Err(PartialVMError::new(
+                StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
```

Comment:
This should also be unreachable? right? Do we need to handle it gracefully?

Reply:
Gracefully as if ..? Here we use invariant violation so that we do not panic, in case something goes wrong